cfgov/unprocessed/apps/youth-employment-success/js/util.js

```diff
@@ -0,0 +1,56 @@
+let UNDEFINED;
+
+const REDUCER_RETURN_ERROR = 'Reducer must return a state object';
+const processStateFromReducers = reducers => (
+  state = UNDEFINED,
+  action = { type: null }
+) => {
+  /**
+   * If reducers is not an object
+   */
+  const reducersKeys = Object.keys( reducers );
+  const finalState = {};
+  let hasChanged = false;
+
+  for ( let i = 0; i < reducersKeys.length; i++ ) {
```

Comment:
Could be `for ( let i = 0, len = reducersKeys.length; i < len; i++ ) {`, though I'm not totally certain that's more efficient.

Reply:
The v8 optimizing compiler, at least, will automatically cache the loop length as a variable when the code. I'm not sure how other compilers treat it, but I think the `.length` property access is probably safe.